### PR TITLE
Clock & Date

### PR DESCRIPTION
**Pull request by onesounds**
https://github.com/Flow-Launcher/Flow.Launcher/discussions/717
https://github.com/Flow-Launcher/Flow.Launcher/issues/1286

image
image

**What's the PR?**
- Add Clock & Date (Time and Cultural info based System Setting) in query area
- It is possible to turn on and off each. (Clock only, Date only, off all)
- Default(when first launch) is Clock only
- Auto Hide when typing or open History/Context menu (this feature is style trigger in xaml only)
- Add Clock and Date Color in themes

**Tests**
- [x] The watch and date should disappear when appropriate.
- [x] Changing time in real time.
- [x] Settings shall be applied immediately.
- [ ] It should not significantly affect the speed of the process. 

**Todo (need help!)**
- [x] The clock code needs to be organized. (mainwindow.cs in now)
- [x] The preview should work on the theme. 

**ETC**
- If someone want like "DDDD HH:MM"... ?
- I don't know how to give the option for user. 
- I currently feel that this option is unnecessary.

### CODE REVIEW

**Comment by Garulf** [comment]
Looks great!

**Comment by taooceros** [comment]
Is the theme changed only because of reformatting except `base.xaml`?

**Comment by onesounds** [comment]
image


- Added Clock/Date Format Combo box
- Added Clock/Date Preview (when Toggled or Changed Format)

**Comment by jjw24** [comment]
I run into high memory usage when this pr was tested along with some others not sure if it's from this pr but let me have another test.

File: Flow.Launcher/MainWindow.xaml [hunk 1]
```diff
@@ -193,6 +193,12 @@
                             </ContextMenu>
                         </TextBox.ContextMenu>
                     </TextBox>
+
+                    <StackPanel x:Name="ClockPanel" Style="{DynamicResource ClockPanel}" IsHitTestVisible="False">
```
Comment (taooceros) [inline Flow.Launcher/MainWindow.xaml:197]: what's ishittestvisible?

File: Flow.Launcher/SettingWindow.xaml.cs [hunk 2]
```diff
@@ -350,6 +351,38 @@ private void ItemSizeChanged(object sender, SizeChangedEventArgs e)
...
+            if (settings.UseDate == true)
+            {
+                DateBox.Visibility = Visibility.Visible;
+                DateBox.Text = System.DateTime.Now.ToString(settings.DateFormat);
+            }
+            else if (settings.UseDate == false)
+            {
+                DateBox.Visibility = Visibility.Collapsed;
+            }
+        }
```
Comment (taooceros) [inline Flow.Launcher/SettingWindow.xaml.cs:384]: mind putting these to viewmodel?

File: Flow.Launcher/MainWindow.xaml (hunk 1 above)
Comment (onesounds) [reply Flow.Launcher/MainWindow.xaml:197]: it make the item non-clickable. in this case, focus the textbox when click clock item.

Comment (taooceros) [reply Flow.Launcher/MainWindow.xaml:197]: that's cool!

**Comment by jjw24** [comment]
@onesounds @taooceros merge Dev please

**Review by jjw24 — APPROVED** [review]

**Pull request merged**